src/+init/setBoundaryCondition.m

```diff
@@ -58,10 +58,12 @@
     % 2 --Specified flux(BCh);
     % 3 --Atmospheric forcing;
     NBCh = 3;
-
     BCh = -20 / 3600;
+    
+    GroundwaterSettings = io.readGroundwaterSettings() % added by Mostafa,
+    GroundwaterCoupling = GroundwaterSettings.GroundwaterCoupling;
     % this should be fixed in issue 183
-    if startsWith(initialLandcoverClass, 'Croplands')
+    if startsWith(initialLandcoverClass, 'Croplands') || GroundwaterCouplingCoupling % modified by Mostafa, for checking GroundwaterCoupling coupling
```

Comment:
```suggestion
    if startsWith(initialLandcoverClass, 'Croplands')
```

Reply:
No need to change this function. Instead, in the main script after `load([OutputPath, 'STEMMUS_SCOPE_state.mat']);`, we can write the `if GroundwaterSettings.GroundwaterCoupling`, then `BoundaryCondition.NBChB = 1`

Reply:
I tried to add the line `if GroundwaterSettings.GroundwaterCoupling` as suggested after `load([OutputPath, 'STEMMUS_SCOPE_state.mat']);`. I think it will work only when BMI is in the update` mode. So, when I was testing, the `GroundwaterCoupling` can not be activated.

 Can we shift the line `if GroundwaterSettings.GroundwaterCoupling` to be after line [317](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/4a34628429f61cda017b89cf0325f488777168e9/src/STEMMUS_SCOPE.m#L317) ? So, it will be easier for me during the testing period.

Reply:
yes, it should be moved outside of the if statement.

Reply:
@MostafaGomaa93 just to let you know that after discussing this with Bart, it turns out your suggestion is correct i.e. moving the if stetmanet after line 317. I made a pull request to fix this see #223. After regenerating the executable file, I will merge the pull request. Nothing needs to be done from your side. Just to inform you about coming changes.

Reply:
@SarahAlidoost Thanks for the update